### PR TITLE
Adds bullet stacks / neuters TG gaming mechanics

### PR DESCRIPTION
**Pull request by ProfessorPopoff**
Well well well... An unexpected appearance, but a welcome one. 

## About The Pull Request
This PR ports Septic's implementation of bullet stacks, which are basically just funny magazines. It works well enough, so what can really be said.
image
As can be seen here, multiple bullets in the same hand. Woah! You can load them into magazines or directly into guns that accept direct loads. Which means...

Shotguns and repeaters will now be taking ammo in directly. No more speedloader/box nonsense. Begone with it! 

In addition to this, I've disabled the ability for players to suck ammo directly into magazines from the floor,.

This PR now repaths speedloaders to be their own path. This is so that revolvers can accept speedloaders without actually just outright taking ammo from a random ammo pack- eliminating the whole point of them.

## Why It's Good For The Game

Weapon balancing. Cool. 
Also the SOUL.

##To do?

- [x] - Fix speedloaders
- [ ] - Figure out last bullet getting evaporated into a runtime on bullet piles
- [ ] - maybe work out pre-baked icons instead of random gen
- [x] - Fix certain guns sucking up all the required ammo from an ammo stack with a single click

## Changelog

add: bullet stacks
del: magazine floor ammo sucking
add: soul

### CODE REVIEW

**Comment by Mannybrado** [comment]
soul.

**Comment by Hekzder** [comment]
soul.

**Comment by TheRealScarHomie** [comment]
SOVL

**Comment by ProfessorPopoff** [comment]
Unfortunately I have to re-enabled mag to mag bullet transfers for now until I find a better solution- As it disabled revolver speedloader functionality

File: mojave/code/datums/components/embedded.dm [hunk 1]
```diff
@@ -0,0 +1,156 @@
+/datum/component/embedded
+	var/datum/injury/injury
```
Comment (ChungusGamer666) [inline mojave/code/datums/components/embedded.dm:2]: how did this get here wtf

Comment (ProfessorPopoff) [reply mojave/code/datums/components/embedded.dm]: a fantastic question

**Comment by ChungusGamer666** [comment]
i'll do this pr for your chief you may close it for now

**Pull request closed without merging**